src/common/position_path.rs

```diff
@@ -75,9 +75,11 @@ impl Iterator for PositionPathIter {
                 // If we previously encountered a side node corresponding to an
                 // invalid path node, we observe that the next valid path node
                 // always pairs with this side node. Once the path and side node
-                // have been paired, we continue to pair path  and side nodes
+                // have been paired, we continue to pair path and side nodes
                 // normally.
-                side = self.current_side_node.take().unwrap_or(side);
+                if self.current_side_node.is_some() {
+                    side = self.current_side_node.take().unwrap()
```

Comment:
```suggestion
                    side = self.current_side_node.take().expect("Reason why this will never happed")
```

Reply:
If the safety of this change isn't clear from the code itself, perhaps I should just leave these lines as is. The idea here is to check if `current_side_node.is_some()` and only take and unwrap it, if it is. Both before and after accomplish this, but I changed it to this because I thought this communicates the logic a bit more clearly. 

But the before explicitly will not throw. By keeping it the way it is, I don't have to worry about the extra safety concerns.

Reply:
Oh wait yeh I was tired when reading this. The check is right above it. I still think it's worth adding an inspect because it helps if someone comes along in the future and refactors. They might remove the check without realizing it's needed for the expect.

Reply:
Oh btw another way to write this that removes the need for an unwrap is:
```rust
if let Some(s) = self.current_side_node.take() {
  side = s
}
```